core/src/main/kotlin/com/gitlab/kordlib/core/entity/User.kt

```diff
@@ -91,22 +87,31 @@ open class User(
         val isAnimated: Boolean get() = data.avatar?.startsWith("a_") ?: false
 
         /**
-         * Gets the avatar url in a supported format, prioritizing [Image.Format.GIF] for animated avatars and [Image.Format.PNG] for others.
+         * Get supported format, prioritizing [Image.Format.GIF] for animated avatars and [Image.Format.PNG] for others.
          */
-        val url: String
+        val format: Image.Format
             get() = when {
-                isAnimated -> getUrl(Image.Format.GIF)
-                else -> getUrl(Image.Format.PNG)
-            } ?: defaultUrl
+                isAnimated -> Image.Format.GIF
+                else -> Image.Format.PNG
+            }
+
+        /**
+         * Gets the avatar url in supported format with default behavior of [format] and size [Image.Size.SIZE_128].
+         */
+        val url: String
+            get() = getUrl()!!
 
         /**
-         * Gets the avatar url in a supported [Image.Format], or returns null if the [format] is not supported.
+         * Gets the avatar url in given [format] and [size], or returns null if the [format] is not supported.
+         *
+         * @param format    The requested image format, defaults to the behavior of [Avatar.format] if not specified.
+         * @param size      The requested image resolution, defaults to [Image.Size.SIZE_128] if not specified.
          */
-        fun getUrl(format: Image.Format): String? {
+        fun getUrl(format: Image.Format = this.format, size: Image.Size = Image.Size.SIZE_128): String? {
```

Comment:
Is mandating a size a good idea? The endpoint seems to works fine without it as well.

We've run into a little issue with nullability here, if `format = this.format` then we can guarantee the return type to not be null. I wonder if a separate overload where only `size` could be specified would be a solution to this.

Reply:
Size when not specified discord send us by assuming desired_size = 128.

The default value of format when not specified is this.format, but user can pass its desired format as argument :)

Reply:
> Size when not specified discord send us by assuming desired_size = 128.

I can't find any documented behavior for this. Additionally, this seems to not be correct.
To give some examples: emojis have a default size of 64, while splash screen have a default width of 128 but a non-standard height and are rendered at size 512 by default.

> The default value of format when not specified is this.format, but user can pass its desired format as argument 

What I was trying to convey is that cases where `format` is the default format but sizes are different should result in a non-null `String` return. 
`avatar.getUrl(size = Image.Size.SIZE_128)` should not return a nullable String, 
while `avatar.getUrl(format = Format.GIF, size = Image.Size.SIZE_128)` should return a nullable String.

Reply:
Apologies, The examples do not apply here since this is purely about User's avatars.
The point about it not being documented behavior stands though.